src/plugins/interactive_setup/server/verification_service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fs from 'fs';
+import path from 'path';
+
+import type { Logger } from 'src/core/server';
+
+import { getDetailedErrorMessage } from './errors';
+import { VerificationCode } from './verification_code';
+
+export class VerificationService {
+  private fname: string;
+
+  constructor(configPath: string, private readonly logger: Logger) {
+    this.fname = path.join(path.dirname(configPath), '.code');
+  }
+
+  public setup() {
+    const verificationCode = new VerificationCode(this.logger);
+
+    try {
+      fs.writeFileSync(this.fname, verificationCode.code);
+      this.logger.debug(`Successfully wrote verification code to ${this.fname}`);
+    } catch (error) {
+      this.logger.error(
```

Comment:
__question:__ shouldn't we record the fact that we couldn't create file in the setup contract? I'm thinking about two potential use cases here:

* If Plugin detects that Kibana is run in non-TTY mode and it couldn't create file with verification code, then maybe we should not enable interactive setup?
* If we decide to still enable interactive setup, we could potentially use this flag in UI to know when _not_ to offer using CLI since it will be useless? And in general, how do we plan to offer using CLI to the user? Via documentation only or also in UI?

Reply:
Good point. Let's limit the number of scenarios we need to support and disable interactive setup mode if we can't write the verification token to disk.

Reply:
I have added instructions to the UI. these are still very basic though and we should add more help text in a follow up.